apps/web/src/pages/checkout/[packSlug].tsx

```diff
@@ -0,0 +1,184 @@
+import {
+  CheckoutMethod,
+  PackAuction,
+  PackStatus,
+  PackType,
+  PublishedPack,
+} from '@algomart/schemas'
+import { GetServerSideProps } from 'next'
+import { useRouter } from 'next/router'
+import useTranslation from 'next-translate/useTranslation'
+import { useEffect } from 'react'
+import { v4 as uuid } from 'uuid'
+
+import { ApiClient } from '@/clients/api-client'
+import { Analytics } from '@/clients/firebase-analytics'
+import { usePaymentProvider } from '@/contexts/payment-context'
+import { Environment } from '@/environment'
+import DefaultLayout from '@/layouts/default-layout'
+import {
+  getAuthenticatedUser,
+  handleUnauthenticatedRedirect,
+} from '@/services/api/auth-service'
+import CheckoutTemplate from '@/templates/checkout-template'
+import { urls } from '@/utils/urls'
+
+export interface CheckoutPageProps {
+  address?: string
+  auctionPackId: string | null
+  currentBid: number | null
+  release: PublishedPack
+}
+
+export default function Checkout({
```

Comment:
Changed the route for the checkout page to be `/checkout/[packSlug]` as opposed to `/checkout?pack=[packSlug]`, since pack is a required field and now that we're using query params for method/step, it reduces complexity to not have to keep adding `pack` back onto the route.